java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java

```diff
@@ -245,8 +245,6 @@ public static Errata addChannelsToErrata(Errata errata,
         //Save the errata
         log.debug("addChannelsToErrata - storing errata");
         storeErrata(errata);
-
-        errata = (Errata) HibernateFactory.reload(errata);
```

Comment:
In the end I decided to rollback the patch. I couldn't reproduce the issue (multiple objects with same identifier in the hibernate session) anymore.

Reply:
It makes sense, yeah. Let's keep it in mind in case it happens again in future.

One last question: did you pushed the rollback already? I still see the changes in place at the moment

Reply:
good point - i forgot to push it. i'll just remove the patch and will force-push.